src/Proto.Cluster/Member/MemberList.cs

```diff
@@ -228,13 +231,13 @@ private void MemberLeave(Member memberThatLeft)
 
                 if (ms.GetAllMembers().Count == 0)
                 {
-                    _memberStrategyByKind.Remove(k);
+                    MemberStrategyByKind.Remove(k);
                 }
             }
 
-            _bannedMembers.Add(memberThatLeft.Id);
+            BannedMembers.Add(memberThatLeft.Id);
             
-            _members.Remove(memberThatLeft.Id);
+            Members.Remove(memberThatLeft.Id);
```

Comment:
As this is an immutable object, Remove only removes the value from the immutable copy.

It needs to be this for every place where members are added or removed
```csharp
Members = Members.Remove(...)
```

Reply:
Aggrege. Changes applied.